docs/source/guides/adaptive_batching.rst

```diff
@@ -3,4 +3,44 @@
 Adaptive Batching
 =================
 
-TODO
+Batching is the term used for combining multiple inputs for submission to processing at the same time. The idea is that processing multiple messages is be faster than processing each individual message one at a time. In practice many ML frameworks have optimizations for processing multiple messages at a time because that how the underlying hardware works in many cases.
+
+.. epigraph::
+    "While serving a TensorFlow model, batching individual model inference requests together can be important for performance. In particular, batching is necessary to unlock the high throughput promised by hardware accelerators such as GPUs."
+    -- `TensorFlow documentation <https://github.com/tensorflow/serving/blob/master/tensorflow_serving/batching/README.md>`_
+
+As an optimization for a real-time service, batching works off of 2 main concepts.
```

Comment:
Let's also explain why we do server-side batching. Server side batching is advantageous because 1) it simplifies the client logic 2) and the server can often times batch more efficiently than the client due to traffic volume.

Reply:
👍